### PR TITLE
Weapon/mechanics

### LINKED ISSUE #14: Player Shooting
Add crosshair and improve the player shooting experience : add muzzle flash and particles

### LINKED ISSUE #15: Weapon Mechanics
Develop a switching method for weapons, considering the player has already 2 weapons.  
Implement a reloading system

### PR DESCRIPTION
**Pull request by claudiarmamede**
* Allow Switching weapons mode.
* Allow picking weapons from the ground.
* Reduce bullets while shooting.
* Gun reload.
* Add flash and particles.

### CODE REVIEW

File: Assets/Scripts/Weapons/FireWeapon.cs [hunk 1]
```diff
@@ -1,10 +1,79 @@
...
+
+        Debug.Log("After reload clip: " + _ammo.getClipValue());
+    }
+
+    public void decresaseAmmo(){
+        Debug.Log("Current clip: " + _ammo.getClipValue());
+        _ammo.setClipValue(_ammo.getClipValue() - 1);
+        Debug.Log("After shoot clip: " + _ammo.getClipValue());
+
+    }
```
Comment (CajoAlbuquerque) [inline Assets/Scripts/Weapons/FireWeapon.cs]: There is a typo on the name of the function xD

File: Assets/Scripts/Player/Controller/PlayerShootController.cs [hunk 2]
```diff
@@ -4,32 +4,36 @@ public class PlayerShootController : Shooter
...
+        if (Physics.Raycast(ray, out hit))
+        {
+            hitPoint = ray.GetPoint(hit.distance);
         }
-        else if(Input.GetButton("Fire2")){
-            shoot(meleeWeapon);
+        else
+        {
+            hitPoint = ray.GetPoint(100/* or some very far distance */);
         }
```
Comment (CajoAlbuquerque) [inline Assets/Scripts/Player/Controller/PlayerShootController.cs]: What is the hitpoint used for?

**Review by CajoAlbuquerque — APPROVED** [review]

File: Assets/Scripts/Weapons/FireWeapon.cs (hunk 1 above)
Comment (BernardoCMoreira) [reply Assets/Scripts/Weapons/FireWeapon.cs]: decresase is the new decrease

File: Assets/Scripts/Player/Controller/PlayerShootController.cs (hunk 2 above)
Comment (BernardoCMoreira) [reply Assets/Scripts/Player/Controller/PlayerShootController.cs]: This was some dead code. We were trying to make the weapon follow the crosshair, but found out that we could simply drag the weapon component into the player main camera

**Review by CajoAlbuquerque — APPROVED** [review]

**Pull request merged**